test/pipeline_test_runner.cc

```diff
@@ -731,18 +733,25 @@ TEST_CASE("PerPhaseTest") { // NOLINT
 
         handler.addObserved(*gs, "rbs-rewrite-tree", [&]() { return nodes->toString(*gs); });
 
-        ast::ParsedFile file;
-        switch (parser) {
-            case realmain::options::Parser::ORIGINAL: {
-                file = testSerialize(*gs, ast::ParsedFile{ast::desugar::node2Tree(ctx, move(nodes)), f.file});
-                break;
-            }
-            case realmain::options::Parser::PRISM: {
-                file = testSerialize(*gs, ast::ParsedFile{ast::prismDesugar::node2Tree(ctx, move(nodes)), f.file});
-                break;
+        auto legacyAST = ast::desugar::node2Tree(ctx, move(nodes));
+
+        ast::ExpressionPtr ast;
+        if (directlyDesugaredTree != nullptr) {
+            auto prismAST = ast::prismDesugar::node2Tree(ctx, move(directlyDesugaredTree));
+
+            if (!legacyAST.exactlyEqual(*gs, prismAST, f.file)) {
+                auto expected = legacyAST.showRawWithLocs(*gs, f.file);
+                auto actual = prismAST.showRawWithLocs(*gs, f.file);
+                CHECK_EQ_DIFF(expected, actual,
+                              fmt::format("Prism desugared tree does not match legacy desugared tree"));
```

Comment:
This isn't asserting that the locations are the same as the original parser though, right? There are two paths here:

1. Prism -> Translator -> Desugar
1. Prism -> DirectDesugar

As a result, I think this is only verifying that the locations are consistent between DirectDesugar and Translator, right?

Reply:
Correct.

We have `prism_regression` tests which test the `Prism -> Translator` stage (ensuring we produce an equivalent syntax tree to the original parser).

Then this PR is checking the `DirectDesugar` stage.

The composition of the two sets of tests covers the pipeline from Ruby source text through to the `ast::ExpressionPtr`.